### PR TITLE
[refactor](pipelineX) add MultiCast operator

### PR DESCRIPTION
**Pull request by Mryange**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

File: be/src/pipeline/exec/multi_cast_data_stream_source.cpp [hunk 1]
```diff
@@ -124,4 +125,45 @@ RuntimeProfile* MultiCastDataStreamerSourceOperator::get_runtime_profile() const
     return _multi_cast_data_streamer->profile();
 }
 
+Status MultiCastDataStreamSourceLocalState::init(RuntimeState* state, LocalStateInfo& info) {
+    RETURN_IF_ERROR(Base::init(state, info));
+    SCOPED_TIMER(profile()->total_time_counter());
```
Comment (Gabriel39) [inline be/src/pipeline/exec/multi_cast_data_stream_source.cpp:130]: Also `SCOPED_TIMER(_open_timer);`

File: be/src/pipeline/exec/multi_cast_data_stream_source.h [hunk 2]
```diff
@@ -83,11 +86,157 @@ class MultiCastDataStreamerSourceOperator final : public OperatorBase,
...
+    Status open(doris::RuntimeState* state) override { return Status::OK(); };
+
+    Status prepare(RuntimeState* state) override { return Status::OK(); }
+
+    Status sink(RuntimeState* state, vectorized::Block* in_block,
+                SourceState source_state) override {
+        auto& local_state =
+                state->get_sink_local_state(id())->cast<MultiCastDataStreamSinkLocalState>();
+        SCOPED_TIMER(local_state.profile()->total_time_counter());
+        if (in_block->rows() > 0 || source_state == SourceState::FINISHED) {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/multi_cast_data_stream_source.h:214]: COUNTER_UPDATE(local_state.rows_input_counter(), (int64_t)in_block->rows());

File: be/src/pipeline/exec/union_source_operator.cpp [hunk 3]
```diff
@@ -103,25 +103,53 @@ Status UnionSourceLocalState::init(RuntimeState* state, LocalStateInfo& info) {
     SCOPED_TIMER(profile()->total_time_counter());
     SCOPED_TIMER(_open_timer);
     auto& p = _parent->cast<Parent>();
-    std::shared_ptr<DataQueue> data_queue = std::make_shared<DataQueue>(p._child_size, _dependency);
-    _shared_state->data_queue.swap(data_queue);
+    // std::shared_ptr<DataQueue> data_queue = std::make_shared<DataQueue>(p._child_size, _dependency);
```
Comment (Gabriel39) [inline be/src/pipeline/exec/union_source_operator.cpp:106]: Delete this

File: be/src/pipeline/exec/union_source_operator.cpp [hunk 4]
```diff
@@ -103,25 +103,53 @@ Status UnionSourceLocalState::init(RuntimeState* state, LocalStateInfo& info) {
...
+    // _shared_state->data_queue.swap(data_queue);
+    if (state->per_fragment_instance_idx() == 0) {
```
Comment (Gabriel39) [inline be/src/pipeline/exec/union_source_operator.cpp:108]: Add some comments here

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Comment by Gabriel39** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**